src/rgw/rgw_metadata.cc

```diff
@@ -830,6 +831,19 @@ int RGWMetadataManager::remove(string& metadata_key)
   if (ret < 0)
     return ret;
 
+  if (store->get_zone().tier_type == "archive") {
```

Comment:
Not crazy about this one. How about having a different handlers for the archive zone and override the default ones after `meta_mgr` is being created in RGWRados if it's archive zone?  In that case, just override the `remove()` method and there's no need for a new `remove_rename()`.

Reply:
ok I will have a look